diff --git a/base/check.h b/base/check.h
index 05f7b3755ab33..c6a1e0c1eac1a 100644
--- a/base/check.h
+++ b/base/check.h
@@ -162,7 +162,7 @@ class BASE_EXPORT NotReachedNoreturnError : public CheckError {
 // calling an out-of-line function instead of a noreturn inline macro prevents
 // compiler optimizations.
 #define CHECK(condition) \
-  UNLIKELY(!(condition)) ? logging::CheckFailure() : EAT_CHECK_STREAM_PARAMS()
+  UNLIKELY(!(condition)) ? ::logging::CheckFailure() : EAT_CHECK_STREAM_PARAMS()
 
 #define CHECK_WILL_STREAM() false
 
