diff --git a/base/path_service_unittest.cc b/base/path_service_unittest.cc
index 4cd5ecb9a07af..11ed4805479f1 100644
--- a/base/path_service_unittest.cc
+++ b/base/path_service_unittest.cc
@@ -140,6 +140,12 @@ TEST_F(PathServiceTest, Get) {
                                                  : &ReturnsValidPath,
                  key);
   }
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+  // In the three Starboard custom directories, DIR_CACHE should always be
+  // valid while DIR_SYSTEM_FONTS and DIR_SYSTEM_FONTS_CONFIGURATION
+  // can be invalid on some platforms.
+  EXPECT_PRED1(ReturnsValidPath, DIR_CACHE);
+#else  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 #if BUILDFLAG(IS_WIN)
   for (int key = PATH_WIN_START + 1; key < PATH_WIN_END; ++key) {
     EXPECT_PRED1(ReturnsValidPath, key);
@@ -159,6 +165,7 @@ TEST_F(PathServiceTest, Get) {
     EXPECT_PRED1(ReturnsValidPath, key);
   }
 #endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 }
 
 // Tests that CheckedGet returns the same path as Get.
