diff --git a/base/base_switches.h b/base/base_switches.h
index cead11f7ef247..093ba75edb3ad 100644
--- a/base/base_switches.h
+++ b/base/base_switches.h
@@ -67,6 +67,10 @@ extern const char kEnableCrashpad[];
 extern const char kSchedulerBoostUrgent[];
 #endif
 
+#if BUILDFLAG(IS_COBALT)
+extern const char kCobaltJniPrefix[];
+#endif
+
 }  // namespace switches
 
 #endif  // BASE_BASE_SWITCHES_H_
