diff --git a/base/path_service.cc b/base/path_service.cc
index ea626f026303b..5db6dea9aee8a 100644
--- a/base/path_service.cc
+++ b/base/path_service.cc
@@ -26,7 +26,9 @@ namespace base {
 
 bool PathProvider(int key, FilePath* result);
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+bool PathProviderStarboard(int key, FilePath* result);
+#elif BUILDFLAG(IS_WIN)
 bool PathProviderWin(int key, FilePath* result);
 #elif BUILDFLAG(IS_APPLE)
 bool PathProviderMac(int key, FilePath* result);
@@ -64,6 +66,14 @@ Provider base_provider = {PathProvider, nullptr,
 #endif
                           true};
 
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+Provider base_provider_starboard = {base::PathProviderStarboard, &base_provider,
+#ifndef NDEBUG
+                                    base::PATH_STARBOARD_START,
+                                    base::PATH_STARBOARD_END,
+#endif
+                                    true};
+#else  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 #if BUILDFLAG(IS_WIN)
 Provider base_provider_win = {
   PathProviderWin,
@@ -119,7 +129,7 @@ Provider base_provider_posix = {
   true
 };
 #endif
-
+#endif  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 
 struct PathData {
   Lock lock;
@@ -129,7 +139,9 @@ struct PathData {
   bool cache_disabled;  // Don't use cache if true;
 
   PathData() : cache_disabled(false) {
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+    providers = &base_provider_starboard;
+#elif BUILDFLAG(IS_WIN)
     providers = &base_provider_win;
 #elif BUILDFLAG(IS_APPLE)
     providers = &base_provider_mac;
