diff --git a/base/profiler/module_cache.cc b/base/profiler/module_cache.cc
index 7911f89e5c2f2..e2be34afdece9 100644
--- a/base/profiler/module_cache.cc
+++ b/base/profiler/module_cache.cc
@@ -64,6 +64,9 @@ ModuleCache::~ModuleCache() {
 }
 
 const ModuleCache::Module* ModuleCache::GetModuleForAddress(uintptr_t address) {
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+  return nullptr;
+#else
   if (const ModuleCache::Module* module = GetExistingModuleForAddress(address))
     return module;
 
@@ -77,6 +80,7 @@ const ModuleCache::Module* ModuleCache::GetModuleForAddress(uintptr_t address) {
   // TODO(https://crbug.com/1131769): Reintroduce DCHECK(result.second) after
   // fixing the issue that is causing it to fail.
   return result.first->get();
+#endif  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 }
 
 std::vector<const ModuleCache::Module*> ModuleCache::GetModules() const {
