diff --git a/base/process/launch_posix.cc b/base/process/launch_posix.cc
index 1c55917b28b22..aa43733653869 100644
--- a/base/process/launch_posix.cc
+++ b/base/process/launch_posix.cc
@@ -47,6 +47,10 @@
 #include "base/trace_event/base_tracing.h"
 #include "build/build_config.h"
 
+#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+#include "base/starboard/linker_stub.h"
+#endif  // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_AIX)
 #include <sys/prctl.h>
 #endif
@@ -142,7 +146,12 @@ struct kernel_sigaction {
 long sys_rt_sigaction(int sig,
                       const struct kernel_sigaction* act,
                       struct kernel_sigaction* oact) {
+#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+  COBALT_LINKER_STUB();
+  return -1;
+#else  // ENABLE_COBALT_HERMETIC_HACKS
   return syscall(SYS_rt_sigaction, sig, act, oact, sizeof(kernel_sigset_t));
+#endif  // ENABLE_COBALT_HERMETIC_HACKS
 }
 
 // This function is intended to be used in between fork() and execve() and will
