diff --git a/base/base_paths.h b/base/base_paths.h
index 28c1643464233..8d119e5500ef6 100644
--- a/base/base_paths.h
+++ b/base/base_paths.h
@@ -10,6 +10,9 @@
 
 #include "build/build_config.h"
 
+#if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
+#include "base/base_paths_starboard.h"
+#else
 #if BUILDFLAG(IS_WIN)
 #include "base/base_paths_win.h"
 #elif BUILDFLAG(IS_APPLE)
@@ -21,6 +24,7 @@
 #if BUILDFLAG(IS_POSIX)
 #include "base/base_paths_posix.h"
 #endif
+#endif  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 
 namespace base {
 
