diff --git a/base/task/current_thread.cc b/base/task/current_thread.cc
index b062ee6f827b7..f222b1c28651c 100644
--- a/base/task/current_thread.cc
+++ b/base/task/current_thread.cc
@@ -149,7 +149,7 @@ MessagePumpForUI* CurrentUIThread::GetMessagePumpForUI() const {
   return static_cast<MessagePumpForUI*>(current_->GetMessagePump());
 }
 
-#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN) && !BUILDFLAG(IS_STARBOARD)
 bool CurrentUIThread::WatchFileDescriptor(
     int fd,
     bool persistent,
