diff --git a/base/trace_event/malloc_dump_provider.cc b/base/trace_event/malloc_dump_provider.cc
index 6e64aedb0fc8c..28ad9e828cf01 100644
--- a/base/trace_event/malloc_dump_provider.cc
+++ b/base/trace_event/malloc_dump_provider.cc
@@ -23,7 +23,9 @@
 #include "base/trace_event/traced_value.h"
 #include "build/build_config.h"
 
-#if BUILDFLAG(IS_APPLE)
+#if BUILDFLAG(IS_STARBOARD)
+#include <stdlib.h>
+#elif BUILDFLAG(IS_APPLE)
 #include <malloc/malloc.h>
 #else
 #include <malloc.h>
@@ -187,7 +189,7 @@ void ReportAppleAllocStats(size_t* total_virtual_size,
 
 #if (BUILDFLAG(USE_PARTITION_ALLOC_AS_MALLOC) && BUILDFLAG(IS_ANDROID)) || \
     (!BUILDFLAG(USE_PARTITION_ALLOC_AS_MALLOC) && !BUILDFLAG(IS_WIN) &&    \
-     !BUILDFLAG(IS_APPLE) && !BUILDFLAG(IS_FUCHSIA))
+     !BUILDFLAG(IS_APPLE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_STARBOARD))
 void ReportMallinfoStats(ProcessMemoryDump* pmd,
                          size_t* total_virtual_size,
                          size_t* resident_size,
@@ -356,7 +358,7 @@ bool MallocDumpProvider::OnMemoryDump(const MemoryDumpArgs& args,
   ReportWinHeapStats(args.level_of_detail, nullptr, &total_virtual_size,
                      &resident_size, &allocated_objects_size,
                      &allocated_objects_count);
-#elif BUILDFLAG(IS_FUCHSIA)
+#elif BUILDFLAG(IS_FUCHSIA) || BUILDFLAG(IS_STARBOARD)
 // TODO(fuchsia): Port, see https://crbug.com/706592.
 #else
   ReportMallinfoStats(/*pmd=*/nullptr, &total_virtual_size, &resident_size,
