diff --git a/base/immediate_crash.h b/base/immediate_crash.h
index 049c9eef225d8..bf3392769e709 100644
--- a/base/immediate_crash.h
+++ b/base/immediate_crash.h
@@ -6,6 +6,9 @@
 #define BASE_IMMEDIATE_CRASH_H_
 
 #include "build/build_config.h"
+#if BUILDFLAG(IS_STARBOARD)
+#include "starboard/common/log.h"  // nogncheck
+#endif
 
 // Crashes in the fastest possible way with no attempt at logging.
 // There are several constraints; see http://crbug.com/664209 for more context.
@@ -41,7 +44,11 @@
 // be removed in followups, so splitting it up like this now makes it easy to
 // land the followups.
 
-#if defined(COMPILER_GCC)
+#if BUILDFLAG(IS_STARBOARD)
+#define IMMEDIATE_CRASH() SB_CHECK(false)
+#define TRAP_SEQUENCE1_() SB_CHECK(false)
+#define TRAP_SEQUENCE2_()
+#elif defined(COMPILER_GCC)
 
 #if BUILDFLAG(IS_NACL)
 
