diff --git a/base/base_switches.cc b/base/base_switches.cc
index 8e0a4855d4200..00c1f165b703c 100644
--- a/base/base_switches.cc
+++ b/base/base_switches.cc
@@ -183,4 +183,8 @@ extern const char kEnableCrashpad[] = "enable-crashpad";
 const char kSchedulerBoostUrgent[] = "scheduler-boost-urgent";
 #endif
 
+#if BUILDFLAG(IS_COBALT)
+const char kCobaltJniPrefix[] = "cobalt-jni-prefix";
+#endif
+
 }  // namespace switches
