diff --git a/base/message_loop/message_pump_for_io.h b/base/message_loop/message_pump_for_io.h
index a1080754f6aec..4acbef3e6223d 100644
--- a/base/message_loop/message_pump_for_io.h
+++ b/base/message_loop/message_pump_for_io.h
@@ -22,7 +22,7 @@
 #elif BUILDFLAG(IS_FUCHSIA)
 #include "base/message_loop/message_pump_fuchsia.h"
 #elif BUILDFLAG(IS_POSIX)
-#include "base/message_loop/message_pump_libevent.h"
+#include "base/message_loop/message_pump_epoll.h"
 #endif
 
 namespace base {
@@ -39,7 +39,7 @@ using MessagePumpForIO = MessagePumpDefault;
 #elif BUILDFLAG(IS_FUCHSIA)
 using MessagePumpForIO = MessagePumpFuchsia;
 #elif BUILDFLAG(IS_POSIX)
-using MessagePumpForIO = MessagePumpLibevent;
+using MessagePumpForIO = MessagePumpEpoll;
 #else
 #error Platform does not define MessagePumpForIO
 #endif
