diff --git a/build/build_config.h b/build/build_config.h
index 6db5d9bca0598..c9616bde1fd4d 100644
--- a/build/build_config.h
+++ b/build/build_config.h
@@ -254,6 +254,60 @@
 #define BUILDFLAG_INTERNAL_IS_OZONE() (0)
 #endif
 
+#if defined(ENABLE_BUILDFLAG_IS_COBALT)
+#define BUILDFLAG_INTERNAL_IS_COBALT() (1)
+#else
+#define BUILDFLAG_INTERNAL_IS_COBALT() (0)
+#endif
+
+#if defined(OS_STARBOARD)
+#define BUILDFLAG_INTERNAL_IS_STARBOARD() (1)
+#else
+#define BUILDFLAG_INTERNAL_IS_STARBOARD() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_IS_COBALT_HERMETIC_BUILD)
+#define BUILDFLAG_INTERNAL_IS_COBALT_HERMETIC_BUILD() (1)
+#else
+#define BUILDFLAG_INTERNAL_IS_COBALT_HERMETIC_BUILD() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_USE_STARBOARD_MEDIA)
+#define BUILDFLAG_INTERNAL_USE_STARBOARD_MEDIA() (1)
+#else
+#define BUILDFLAG_INTERNAL_USE_STARBOARD_MEDIA() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_USE_EVERGREEN)
+#define BUILDFLAG_INTERNAL_USE_EVERGREEN() (1)
+#else
+#define BUILDFLAG_INTERNAL_USE_EVERGREEN() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_IS_MODULAR_BUILD)
+#define BUILDFLAG_INTERNAL_MODULAR_BUILD() (1)
+#else
+#define BUILDFLAG_INTERNAL_MODULAR_BUILD() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_COBALT_IS_RELEASE_BUILD)
+#define BUILDFLAG_INTERNAL_COBALT_IS_RELEASE_BUILD() (1)
+#else
+#define BUILDFLAG_INTERNAL_COBALT_IS_RELEASE_BUILD() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_ENABLE_COBALT_HERMETIC_HACKS)
+#define BUILDFLAG_INTERNAL_ENABLE_COBALT_HERMETIC_HACKS() (1)
+#else
+#define BUILDFLAG_INTERNAL_ENABLE_COBALT_HERMETIC_HACKS() (0)
+#endif
+
+#if defined(ENABLE_BUILDFLAG_IS_ANDROIDTV)
+#define BUILDFLAG_INTERNAL_IS_ANDROIDTV() (1)
+#else
+#define BUILDFLAG_INTERNAL_IS_ANDROIDTV() (0)
+#endif
+
 // Compiler detection. Note: clang masquerades as GCC on POSIX and as MSVC on
 // Windows.
 #if defined(__GNUC__)
