diff --git a/build/toolchain/android/BUILD.gn b/build/toolchain/android/BUILD.gn
index 20257d9ace193..e2850e18b3031 100644
--- a/build/toolchain/android/BUILD.gn
+++ b/build/toolchain/android/BUILD.gn
@@ -95,9 +95,17 @@ android_clang_toolchain("android_clang_x86") {
 }
 
 android_clang_toolchain("android_clang_arm") {
+  if (is_cobalt) {
+    toolchain_args = {
+      current_cpu = "arm"
+      starboard_toolchain = "//build/toolchain/android:android_clang_arm"
+      target_platform = "android-arm"
+    }
+  } else {
   toolchain_args = {
     current_cpu = "arm"
   }
+  }
 }
 
 android_clang_toolchain("android_clang_mipsel") {
@@ -131,9 +139,17 @@ android_clang_toolchain("android_clang_x64") {
 }
 
 android_clang_toolchain("android_clang_arm64") {
+  if (is_cobalt) {
+    toolchain_args = {
+      current_cpu = "arm64"
+      starboard_toolchain = "//build/toolchain/android:android_clang_arm64"
+      target_platform = "android-arm64"
+    }
+  } else {
   toolchain_args = {
     current_cpu = "arm64"
   }
+  }
 }
 
 android_clang_toolchain("android_clang_arm64_hwasan") {
