diff --git a/BUILD.gn b/BUILD.gn
index 27ef75902d11e..da992bf83efb1 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -838,6 +838,9 @@ group("gn_all") {
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
+  if (is_cobalt) {
+    deps += [ "//cobalt:gn_all" ]
+  }
 }
 
 # All Rust targets. This is provided for convenience while developing Rust to
@@ -1455,6 +1458,12 @@ if (!is_ios && !is_cronet_build) {
         "//third_party/blink/web_tests/SmokeTests/",
       ]
     }
+
+    if (is_cobalt) {
+      data += [
+        "//third_party/blink/web_tests/wpt_internal/cobalt",
+      ]
+    }
   }
 
   script_test("blink_wpt_tests") {
