diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index dea380727e732..110defa736f63 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -24,6 +24,12 @@ declare_args() {
                 current_cpu == "mipsel" || current_cpu == "mips64el"
 }
 
+if (is_cobalt_hermetic_build) {
+  # We want to avoid adding the sysroot and bringing in unneeded
+  # dependencies in cobalt.
+  use_sysroot = use_sysroot && current_toolchain != default_toolchain
+}
+
 if (sysroot == "") {
   if (current_os == target_os && current_cpu == target_cpu &&
       target_sysroot != "") {
