diff --git a/build/config/android/rules.gni b/build/config/android/rules.gni
index a3eccbff751c0..64d03b27be005 100644
--- a/build/config/android/rules.gni
+++ b/build/config/android/rules.gni
@@ -2299,11 +2299,24 @@ if (enable_java_templates && is_android) {
             "_RESOURCES_VERSION_VARIABLE=${invoker.resources_version_variable}",
           ]
         }
+      } else if (is_cronet_build && !is_java_debug) {
+        # Cronet never uses use_final_fields=true, so make this a reasonable default.
+        defines += [ "_DISABLE_DEBUG_LOGS" ]
       }
 
-      if (defined(testonly) && testonly) {
+      _test_only = defined(testonly) && testonly
+      if (_test_only) {
         defines += [ "_IS_FOR_TEST" ]
       }
+
+      if (!is_java_debug && !_test_only) {
+        defines += [ "_DISABLE_DEBUG_LOGS" ]
+      }
+
+      # TODO(cobalt, b/375655377): Remove once Cobalt build targets remove test_only tag.
+      if (is_cobalt) {
+        defines += [ "_DISABLE_DEBUG_LOGS" ]
+      }
     }
   }
 
@@ -4556,6 +4569,13 @@ if (enable_java_templates && is_android) {
 
     _use_scanned_assets = !_ignore_assets && _scanned_files.assets != []
     _has_resources = _scanned_files.resources != []
+    _common_deps = [ ":$_unpack_target_name" ]
+    if (defined(invoker.deps)) {
+      _common_deps += invoker.deps
+    }
+    if (defined(invoker.public_deps)) {
+      _common_deps += invoker.public_deps
+    }
 
     assert(_ignore_aidl || _scanned_files.aidl == [],
            "android_aar_prebuilt() aidl not yet supported." +
@@ -4632,7 +4652,7 @@ if (enable_java_templates && is_android) {
                                  "testonly",
                                  "strip_drawables",
                                ])
-        public_deps = [ ":$_unpack_target_name" ]
+        deps = _common_deps
         if (_should_process_manifest) {
           android_manifest_dep = ":$_unpack_target_name"
           android_manifest = "${_output_path}/AndroidManifest.xml"
@@ -4702,7 +4722,7 @@ if (enable_java_templates && is_android) {
         _subjar_targets += [ ":$_current_target" ]
         java_prebuilt(_current_target) {
           forward_variables_from(invoker, _java_library_vars)
-          deps = [ ":$_unpack_target_name" ]
+          deps = _common_deps
           if (!defined(requires_android)) {
             requires_android = true
           }
@@ -4719,15 +4739,11 @@ if (enable_java_templates && is_android) {
         forward_variables_from(invoker, _java_library_vars)
         forward_variables_from(invoker,
                                [
-                                 "deps",
                                  "input_jars_paths",
                                  "mergeable_android_manifests",
                                  "proguard_configs",
                                ])
-        if (!defined(deps)) {
-          deps = []
-        }
-        deps += _subjar_targets + [ ":$_unpack_target_name" ]
+        deps = _common_deps + _subjar_targets
         if (defined(_res_target_name)) {
           deps += [ ":$_res_target_name" ]
         }
