diff --git a/build/config/android/config.gni b/build/config/android/config.gni
index 5f4836726a3ea..3d5731bce5936 100644
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -43,7 +43,11 @@ if (is_android || is_chromeos) {
 
     # The default to use for android:minSdkVersion for targets that do
     # not explicitly set it.
-    default_min_sdk_version = 24
+    if (is_cobalt) {
+      default_min_sdk_version = 26
+    } else {
+      default_min_sdk_version = 24
+    }
 
     # Static analysis can be either "on" or "off" or "build_server". This
     # controls how android lint, error-prone, bytecode checks are run. This
@@ -87,8 +91,13 @@ if (is_android || is_chromeos) {
 
   if (!defined(default_android_ndk_root)) {
     default_android_ndk_root = "//third_party/android_ndk"
-    default_android_ndk_version = "r23"
-    default_android_ndk_major_version = 23
+    if (is_cobalt) {
+      default_android_ndk_version = "r26"
+      default_android_ndk_major_version = 26
+    } else {
+      default_android_ndk_version = "r23"
+      default_android_ndk_major_version = 23
+    }
   } else {
     assert(defined(default_android_ndk_version))
     assert(defined(default_android_ndk_major_version))
