diff --git a/build/config/ozone_extra.gni b/build/config/ozone_extra.gni
index cbbca1efe223b..7b8657d9f12bb 100644
--- a/build/config/ozone_extra.gni
+++ b/build/config/ozone_extra.gni
@@ -31,3 +31,15 @@ ozone_external_platform_ui_test_support_deps = []
 # source_set can be listed here so that they can included into
 # interactive_ui_tests.
 ozone_external_interactive_ui_tests_deps = []
+
+if (is_starboard) {
+  ozone_external_platforms += ["starboard"]
+  ozone_external_platform_deps += ["platform/starboard"]
+  ozone_external_platform_test_deps += ["platform/starboard:starboard_unittests"]
+  ozone_external_platform_ui_test_support_deps += ["platform/starboard:test_support"]
+}
+
+if (is_cobalt && target_os != "android" && current_toolchain == host_toolchain) {
+  ozone_external_platforms += [ "headless" ]
+  ozone_external_platform_deps += [ "platform/headless" ]
+}
