diff --git a/content/shell/android/BUILD.gn b/content/shell/android/BUILD.gn
index eb50661286a12..396a6c233d0ab 100644
--- a/content/shell/android/BUILD.gn
+++ b/content/shell/android/BUILD.gn
@@ -22,7 +22,6 @@ shared_library("libcontent_shell_content_view") {
   testonly = true
   deps = [
     ":content_shell_jni_headers",
-    "//components/crash/content/browser",
     "//content/shell:content_shell_app",
     "//content/shell:content_shell_lib",
     "//content/shell:pak",
@@ -30,6 +29,12 @@ shared_library("libcontent_shell_content_view") {
     "//skia",
   ]
 
+  if (!is_cobalt) {
+    deps += [
+      "//components/crash/content/browser",
+    ]
+  }
+
   # Explicit dependency required for JNI registration to be able to
   # find the native side functions.
   if (is_component_build) {
@@ -191,8 +196,6 @@ template("content_shell_apk_tmpl") {
       ":content_shell_assets",
       ":content_shell_java",
       "//base:base_java_test_support",
-      "//components/crash/android:java",
-      "//components/crash/core/app:chrome_crashpad_handler_named_as_so",
       "//components/metrics:metrics_java",
       "//content/public/android:content_java",
       "//content/public/test/android:android_test_message_pump_support_java",
@@ -202,7 +205,15 @@ template("content_shell_apk_tmpl") {
       "//third_party/mesa_headers",
       "//ui/android:ui_java",
     ]
-    loadable_modules = [ "$root_out_dir/libchrome_crashpad_handler.so" ]
+
+    if (!is_cobalt) {
+      deps += [
+        "//components/crash/android:java",
+        "//components/crash/core/app:chrome_crashpad_handler_named_as_so",
+      ]
+
+      loadable_modules = [ "$root_out_dir/libchrome_crashpad_handler.so" ]
+    }
   }
 }
 
