diff --git a/content/public/browser/url_data_source.cc b/content/public/browser/url_data_source.cc
index 56c622745d864..b0965483d9b66 100644
--- a/content/public/browser/url_data_source.cc
+++ b/content/public/browser/url_data_source.cc
@@ -113,6 +113,9 @@ std::string URLDataSource::GetContentSecurityPolicy(
     case network::mojom::CSPDirectiveName::WorkerSrc:
     case network::mojom::CSPDirectiveName::ReportTo:
     case network::mojom::CSPDirectiveName::NavigateTo:
+#if BUILDFLAG(IS_COBALT)
+    case network::mojom::CSPDirectiveName::CobaltLocationSrc:
+#endif
     case network::mojom::CSPDirectiveName::Unknown:
       return std::string();
   }
