diff --git a/content/browser/browser_child_process_host_impl.cc b/content/browser/browser_child_process_host_impl.cc
index 637eb299f8ad1..ed713c6d519a9 100644
--- a/content/browser/browser_child_process_host_impl.cc
+++ b/content/browser/browser_child_process_host_impl.cc
@@ -390,7 +390,7 @@ ChildProcessTerminationInfo BrowserChildProcessHostImpl::GetTerminationInfo(
     // If the delegate doesn't use Launch() helper.
     ChildProcessTerminationInfo info;
     // TODO(crbug.com/1412835): iOS is single process mode for now.
-#if !BUILDFLAG(IS_IOS)
+#if !BUILDFLAG(IS_IOS) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
     info.status = base::GetTerminationStatus(data_.GetProcess().Handle(),
                                              &info.exit_code);
 #endif
diff --git a/content/browser/child_process_launcher_helper_starboard.cc b/content/browser/child_process_launcher_helper_starboard.cc
new file mode 100644
index 0000000000000..021a7703ece95
