diff --git a/content/renderer/content_security_policy_util.cc b/content/renderer/content_security_policy_util.cc
index 5cb6412d6d172..5c2f75325a433 100644
--- a/content/renderer/content_security_policy_util.cc
+++ b/content/renderer/content_security_policy_util.cc
@@ -51,7 +51,11 @@ network::mojom::CSPSourceListPtr BuildCSPSourceList(
       source_list.allow_inline_speculation_rules, source_list.allow_eval,
       source_list.allow_wasm_eval, source_list.allow_wasm_unsafe_eval,
       source_list.allow_dynamic, source_list.allow_unsafe_hashes,
+#if BUILDFLAG(IS_COBALT)
+      source_list.report_sample, source_list.cobalt_insecure_local_network);
+#else
       source_list.report_sample);
+#endif
 }
 
 blink::WebVector<blink::WebString> ToWebVectorOfWebStrings(
@@ -98,7 +102,12 @@ blink::WebCSPSourceList ToWebCSPSourceList(
           source_list->allow_wasm_unsafe_eval,
           source_list->allow_dynamic,
           source_list->allow_unsafe_hashes,
+#if BUILDFLAG(IS_COBALT)
+          source_list->report_sample,
+          source_list->cobalt_insecure_local_network};
+#else
           source_list->report_sample};
+#endif
 }
 
 absl::optional<blink::WebCSPTrustedTypes> ToOptionalWebCSPTrustedTypes(
