diff --git a/components/viz/common/features.cc b/components/viz/common/features.cc
index ebc18015f2812..a54b460cb8291 100644
--- a/components/viz/common/features.cc
+++ b/components/viz/common/features.cc
@@ -35,7 +35,7 @@ namespace features {
 
 BASE_FEATURE(kUseMultipleOverlays,
              "UseMultipleOverlays",
-#if BUILDFLAG(IS_CHROMEOS_ASH)
+#if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(USE_STARBOARD_MEDIA)
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
              base::FEATURE_DISABLED_BY_DEFAULT
