diff --git a/content/public/common/BUILD.gn b/content/public/common/BUILD.gn
index 6d88bcb5c1915..ca8e37596484c 100644
--- a/content/public/common/BUILD.gn
+++ b/content/public/common/BUILD.gn
@@ -97,6 +97,10 @@ source_set("static_switches") {
 # above, this is for use inside of content/ only and does not link statically.
 source_set("switches") {
   visibility = [ "//content/*" ]
+  if (is_cobalt) {
+    # Allows cobalt and cobalt_unittests to access switch definitions.
+    visibility += [ "//cobalt:*" ]
+  }
   public = [ "content_switches.h" ]
   sources = [
     "//content/common/content_export.h",
