diff --git a/content/utility/BUILD.gn b/content/utility/BUILD.gn
index a717b924d9bc5..f54a8165e52c7 100644
--- a/content/utility/BUILD.gn
+++ b/content/utility/BUILD.gn
@@ -102,6 +102,12 @@ source_set("utility") {
       "//media/mojo/services",
       "//third_party/angle:angle_gpu_info_util",
     ]
+    if (is_cobalt_hermetic_build) {
+      deps -= [
+        "//gpu/config",
+        "//third_party/angle:angle_gpu_info_util",
+      ]
+    }
   }
 
   if (is_chromeos_ash && is_chrome_branded) {
