diff --git a/content/renderer/pepper/plugin_module.cc b/content/renderer/pepper/plugin_module.cc
index 127b5cd54a197..2352fc01d66cd 100644
--- a/content/renderer/pepper/plugin_module.cc
+++ b/content/renderer/pepper/plugin_module.cc
@@ -331,6 +331,10 @@ const void* GetInterface(const char* name) {
 // given structure. Returns true on success.
 bool LoadEntryPointsFromLibrary(const base::NativeLibrary& library,
                                 ContentPluginInfo::EntryPoints* entry_points) {
+// TODO: (cobalt b/409755808) Try to turn off pepper entirely with enable_ppapi=false.
+#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+  return false;
+#else
   entry_points->get_interface =
       reinterpret_cast<ContentPluginInfo::GetInterfaceFunc>(
           base::GetFunctionPointerFromNativeLibrary(library,
@@ -357,6 +361,7 @@ bool LoadEntryPointsFromLibrary(const base::NativeLibrary& library,
                                                     "PPP_ShutdownModule"));
 
   return true;
+#endif
 }
 
 void CreateHostForInProcessModule(RenderFrameImpl* render_frame,
@@ -422,8 +427,10 @@ PluginModule::~PluginModule() {
   if (entry_points_.shutdown_module)
     entry_points_.shutdown_module();
 
+#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
   if (library_)
     base::UnloadNativeLibrary(library_);
+#endif
 
   // Notifications that we've been deleted should be last.
   HostGlobals::Get()->ModuleDeleted(pp_module_);
@@ -462,7 +469,9 @@ bool PluginModule::InitAsLibrary(const base::FilePath& path) {
 
   if (!LoadEntryPointsFromLibrary(library, &entry_points) ||
       !InitializeModule(entry_points)) {
+#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
     base::UnloadNativeLibrary(library);
+#endif
     return false;
   }
   entry_points_ = entry_points;
