diff --git a/content/public/test/network_service_test_helper.cc b/content/public/test/network_service_test_helper.cc
index f371ea9ad5305..c8ceb5990349e 100644
--- a/content/public/test/network_service_test_helper.cc
+++ b/content/public/test/network_service_test_helper.cc
@@ -758,7 +758,7 @@ class NetworkServiceTestHelper::NetworkServiceTestImpl
     system_task_ptr->Start(std::move(results_cb));
   }
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
   void GetAddressMapCacheLinux(
       GetAddressMapCacheLinuxCallback callback) override {
     const net::AddressMapOwnerLinux* address_map_owner =
@@ -766,7 +766,7 @@ class NetworkServiceTestHelper::NetworkServiceTestImpl
     std::move(callback).Run(address_map_owner->GetAddressMap(),
                             address_map_owner->GetOnlineLinks());
   }
-#endif  // BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 
  private:
   void OnMemoryPressure(
