diff --git a/content/web_test/browser/web_test_content_browser_client.h b/content/web_test/browser/web_test_content_browser_client.h
index 22254206063ab..6dfd1990d7949 100644
--- a/content/web_test/browser/web_test_content_browser_client.h
+++ b/content/web_test/browser/web_test_content_browser_client.h
@@ -25,6 +25,10 @@
 #include "third_party/blink/public/mojom/permissions/permission_automation.mojom-forward.h"
 #include "third_party/blink/public/mojom/storage_access/storage_access_automation.mojom-forward.h"
 
+#if BUILDFLAG(IS_COBALT)
+#include "cobalt/browser/h5vcc_runtime/public/mojom/h5vcc_runtime.mojom-forward.h"
+#endif
+
 namespace blink {
 namespace web_pref {
 struct WebPreferences;
@@ -39,6 +43,10 @@ class MockBadgeService;
 class MockClipboardHost;
 class WebTestBrowserContext;
 
+#if BUILDFLAG(IS_COBALT)
+class StubH5vccRuntimeImpl;
+#endif
+
 class WebTestContentBrowserClient : public ShellContentBrowserClient {
  public:
   // Gets the current instance.
@@ -156,6 +164,12 @@ class WebTestContentBrowserClient : public ShellContentBrowserClient {
       int render_process_id,
       mojo::PendingAssociatedReceiver<mojom::WebTestControlHost> receiver);
 
+#if BUILDFLAG(IS_COBALT)
+  void BindH5vccRuntime(
+      RenderFrameHost* render_frame_host,
+      mojo::PendingReceiver<h5vcc_runtime::mojom::H5vccRuntime> receiver);
+#endif
+
   bool block_popups_ = true;
   bool screen_orientation_changed_ = false;
 
@@ -168,6 +182,10 @@ class WebTestContentBrowserClient : public ShellContentBrowserClient {
       cookie_managers_;
   mojo::UniqueReceiverSet<blink::test::mojom::AttributionReportingAutomation>
       attribution_reporting_receivers_;
+
+#if BUILDFLAG(IS_COBALT)
+  std::unique_ptr<StubH5vccRuntimeImpl> stub_h5vcc_runtime_impl_;
+#endif
 };
 
 }  // namespace content
