diff --git a/content/browser/gpu/gpu_data_manager_testing_arrays_and_structs_autogen.h b/content/browser/gpu/gpu_data_manager_testing_arrays_and_structs_autogen.h
index 749944538ba82..0466d48ab10d9 100644
--- a/content/browser/gpu/gpu_data_manager_testing_arrays_and_structs_autogen.h
+++ b/content/browser/gpu/gpu_data_manager_testing_arrays_and_structs_autogen.h
@@ -91,8 +91,8 @@ const GpuControlList::More kMoreForEntry3_572251052 = {
 };
 
 const int kFeatureListForGpuManagerTestingEntry4[2] = {
-    GPU_FEATURE_TYPE_ACCELERATED_WEBGL2,
     GPU_FEATURE_TYPE_ACCELERATED_WEBGL,
+    GPU_FEATURE_TYPE_ACCELERATED_WEBGL2,
 };
 
 const GpuControlList::GLStrings kGLStringsForGpuManagerTestingEntry4 = {
@@ -171,18 +171,19 @@ const GpuControlList::More kMoreForEntry5_572251052Exception0 = {
     GpuControlList::kDontCare,  // subpixel_font_rendering
 };
 
-const int kFeatureListForGpuManagerTestingEntry6[11] = {
+const int kFeatureListForGpuManagerTestingEntry6[12] = {
+    GPU_FEATURE_TYPE_ACCELERATED_2D_CANVAS,
+    GPU_FEATURE_TYPE_ACCELERATED_WEBGL,
+    GPU_FEATURE_TYPE_ACCELERATED_VIDEO_DECODE,
+    GPU_FEATURE_TYPE_ACCELERATED_VIDEO_ENCODE,
+    GPU_FEATURE_TYPE_GPU_RASTERIZATION,
     GPU_FEATURE_TYPE_ACCELERATED_WEBGL2,
-    GPU_FEATURE_TYPE_ACCELERATED_WEBGPU,
     GPU_FEATURE_TYPE_ANDROID_SURFACE_CONTROL,
-    GPU_FEATURE_TYPE_GPU_RASTERIZATION,
-    GPU_FEATURE_TYPE_CANVAS_OOP_RASTERIZATION,
-    GPU_FEATURE_TYPE_ACCELERATED_2D_CANVAS,
     GPU_FEATURE_TYPE_ACCELERATED_GL,
-    GPU_FEATURE_TYPE_ACCELERATED_VIDEO_DECODE,
-    GPU_FEATURE_TYPE_ACCELERATED_WEBGL,
     GPU_FEATURE_TYPE_VULKAN,
-    GPU_FEATURE_TYPE_ACCELERATED_VIDEO_ENCODE,
+    GPU_FEATURE_TYPE_CANVAS_OOP_RASTERIZATION,
+    GPU_FEATURE_TYPE_ACCELERATED_WEBGPU,
+    GPU_FEATURE_TYPE_SKIA_GRAPHITE,
 };
 
 const GpuControlList::More kMoreForEntry6_572251052 = {
