diff --git a/content/browser/renderer_host/render_process_host_impl.cc b/content/browser/renderer_host/render_process_host_impl.cc
index b6352bba532f2..da15578a7fc8a 100644
--- a/content/browser/renderer_host/render_process_host_impl.cc
+++ b/content/browser/renderer_host/render_process_host_impl.cc
@@ -1603,8 +1603,12 @@ void RenderProcessHostImpl::ShutDownInProcessRenderer() {
     case 1: {
       RenderProcessHostImpl* host = static_cast<RenderProcessHostImpl*>(
           AllHostsIterator().GetCurrentValue());
-      for (auto& observer : host->observers_)
+      for (auto& observer : host->observers_) {
+        observer.InProcessRendererExiting(host);
+      }
+      for (auto& observer : host->observers_) {
         observer.RenderProcessHostDestroyed(host);
+      }
 #ifndef NDEBUG
       host->is_self_deleted_ = true;
 #endif
@@ -5190,7 +5194,9 @@ void RenderProcessHostImpl::OnProcessLaunched() {
       // Not all platforms launch processes in the same backgrounded state. Make
       // sure |priority_.visible| reflects this platform's initial process
       // state.
-#if BUILDFLAG(IS_APPLE)
+#if BUILDFLAG(IS_STARBOARD)
+    priority_.visible = true;
+#elif BUILDFLAG(IS_APPLE)
     priority_.visible =
         !child_process_launcher_->GetProcess().IsProcessBackgrounded(
             ChildProcessTaskPortProvider::GetInstance());
