diff --git a/content/renderer/media/media_factory.h b/content/renderer/media/media_factory.h
index f0fab474c01de..fdbae2d6d739c 100644
--- a/content/renderer/media/media_factory.h
+++ b/content/renderer/media/media_factory.h
@@ -131,7 +131,12 @@ class MediaFactory {
       const RenderFrameMediaPlaybackOptions& renderer_media_playback_options,
       media::DecoderFactory* decoder_factory,
       std::unique_ptr<media::RemotePlaybackClientWrapper> client_wrapper,
-      base::WeakPtr<media::MediaObserver>* out_media_observer);
+      base::WeakPtr<media::MediaObserver>* out_media_observer,
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+      int element_id, const std::string& max_video_capabilities);
+#else // BUILDFLAG(USE_STARBOARD_MEDIA)
+      int element_id);
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
 
   blink::WebMediaPlayer* CreateWebMediaPlayerForMediaStream(
       blink::WebMediaPlayerClient* client,
