diff --git a/components/viz/service/BUILD.gn b/components/viz/service/BUILD.gn
index f2fe76f53dac7..1875cbd89759d 100644
--- a/components/viz/service/BUILD.gn
+++ b/components/viz/service/BUILD.gn
@@ -11,6 +11,10 @@ import("//media/gpu/args.gni")
 import("//skia/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 config("viz_service_implementation") {
 }
 
@@ -341,6 +345,17 @@ viz_component("service") {
     ]
   }
 
+  if (is_cobalt) {
+    if (use_starboard_media) {
+      sources += [
+        "display/starboard/overlay_strategy_underlay_starboard.cc",
+        "display/starboard/overlay_strategy_underlay_starboard.h",
+      ]
+
+      deps += [ "//cobalt/media/service/mojom" ]
+    }
+  }
+
   if (enable_cast_overlay_strategy) {
     sources += [
       "display/overlay_strategy_underlay_cast.cc",
