diff --git a/fuchsia_web/webengine/renderer/web_engine_content_renderer_client.h b/fuchsia_web/webengine/renderer/web_engine_content_renderer_client.h
index de05707ed94b2..efadedda67503 100644
--- a/fuchsia_web/webengine/renderer/web_engine_content_renderer_client.h
+++ b/fuchsia_web/webengine/renderer/web_engine_content_renderer_client.h
@@ -76,7 +76,8 @@ class WebEngineContentRendererClient : public content::ContentRendererClient {
       media::MediaLog* media_log,
       media::DecoderFactory* decoder_factory,
       base::RepeatingCallback<media::GpuVideoAcceleratorFactories*()>
-          get_gpu_factories_cb) override;
+          get_gpu_factories_cb,
+      int element_id) override;
 
 #if BUILDFLAG(ENABLE_CAST_RECEIVER)
   std::unique_ptr<cast_streaming::ResourceProvider>
