diff --git a/gin/BUILD.gn b/gin/BUILD.gn
index 76c024fc5da86..151891350169a 100644
--- a/gin/BUILD.gn
+++ b/gin/BUILD.gn
@@ -153,7 +153,7 @@ if (is_android && enable_java_templates &&
         sources = [ "$root_out_dir/v8_context_snapshot_32.bin" ]
         deps =
             [ "//tools/v8_context_snapshot($android_secondary_abi_toolchain)" ]
-      } else {
+      } else if (v8_use_external_startup_data) {
         _secondary_abi_out_dir =
             get_label_info("//v8($android_secondary_abi_toolchain)",
                            "root_out_dir")
diff --git a/gpu/command_buffer/service/starboard/starboard_surface_texture_gl_owner.cc b/gpu/command_buffer/service/starboard/starboard_surface_texture_gl_owner.cc
new file mode 100644
index 0000000000000..562ae6c6acd26
