diff --git a/gpu/ipc/service/gpu_memory_buffer_factory_native_pixmap.h b/gpu/ipc/service/gpu_memory_buffer_factory_native_pixmap.h
index 9e9d71908cb7f..6c1630aed6563 100644
--- a/gpu/ipc/service/gpu_memory_buffer_factory_native_pixmap.h
+++ b/gpu/ipc/service/gpu_memory_buffer_factory_native_pixmap.h
@@ -5,8 +5,6 @@
 #ifndef GPU_IPC_SERVICE_GPU_MEMORY_BUFFER_FACTORY_NATIVE_PIXMAP_H_
 #define GPU_IPC_SERVICE_GPU_MEMORY_BUFFER_FACTORY_NATIVE_PIXMAP_H_
 
-#include <vulkan/vulkan_core.h>
-
 #include <unordered_map>
 #include <utility>
 
@@ -16,6 +14,12 @@
 #include "gpu/ipc/service/gpu_memory_buffer_factory.h"
 #include "ui/gfx/native_pixmap.h"
 
+// TODO: (cobalt b/409766462): Exclude this file from the build entirely.
+#include "build/build_config.h"
+#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+#include <vulkan/vulkan_core.h>
+#endif
+
 namespace gpu {
 
 class VulkanDeviceQueue;
