diff --git a/infra/config/subprojects/chromium/ci/chromium.gpu.star b/infra/config/subprojects/chromium/ci/chromium.gpu.star
index 789fba7c7ceaa..ca04175b7d4c5 100644
--- a/infra/config/subprojects/chromium/ci/chromium.gpu.star
+++ b/infra/config/subprojects/chromium/ci/chromium.gpu.star
@@ -74,10 +74,6 @@ ci.gpu.linux_builder(
             config = "chromium",
             apply_configs = [
                 "use_clang_coverage",
-                # This is not necessary for this builder itself, but is
-                # necessary in order to keep configs in sync with
-                # "ci/Linux Builder" in order for mirroring to work correctly.
-                "chromium_with_telemetry_dependencies",
             ],
         ),
         chromium_config = builder_config.chromium_config(
@@ -126,10 +122,6 @@ ci.gpu.mac_builder(
         gclient_config = builder_config.gclient_config(
             config = "chromium",
             apply_configs = [
-                # This is not necessary for this builder itself, but is
-                # necessary in order to keep configs in sync with
-                # "ci/Mac Builder" in order for mirroring to work correctly.
-                "chromium_with_telemetry_dependencies",
                 "use_clang_coverage",
             ],
         ),
@@ -181,10 +173,6 @@ ci.gpu.windows_builder(
         gclient_config = builder_config.gclient_config(
             config = "chromium",
             apply_configs = [
-                # This is not necessary for this builder itself, but is
-                # necessary in order to keep configs in sync with
-                # "ci/Mac Builder" in order for mirroring to work correctly.
-                "chromium_with_telemetry_dependencies",
                 "use_clang_coverage",
             ],
         ),
