diff --git a/infra/config/subprojects/chromium/ci/chromium.linux.star b/infra/config/subprojects/chromium/ci/chromium.linux.star
index 78c8f4ba474b5..f5b4f1c1eaa90 100644
--- a/infra/config/subprojects/chromium/ci/chromium.linux.star
+++ b/infra/config/subprojects/chromium/ci/chromium.linux.star
@@ -206,9 +206,6 @@ ci.builder(
             config = "chromium",
             apply_configs = [
                 "use_clang_coverage",
-                # This is necessary due to child builders running the
-                # telemetry_perf_unittests suite.
-                "chromium_with_telemetry_dependencies",
             ],
         ),
         chromium_config = builder_config.chromium_config(
