diff --git a/media/mojo/mojom/BUILD.gn b/media/mojo/mojom/BUILD.gn
index 1214f486e5c62..4297fb9e2dae0 100644
--- a/media/mojo/mojom/BUILD.gn
+++ b/media/mojo/mojom/BUILD.gn
@@ -5,6 +5,10 @@
 import("//media/media_options.gni")
 import("//mojo/public/tools/bindings/mojom.gni")
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 mojom("mojom") {
   generate_java = true
 
@@ -74,6 +78,10 @@ mojom("mojom") {
     ]
   }
 
+  if (is_cobalt && use_starboard_media) {
+    sources += [ "starboard/starboard_media_types.mojom" ]
+  }
+
   public_deps = [
     ":audio_data",
     ":encryption_pattern",
@@ -123,6 +131,10 @@ mojom("mojom") {
     enabled_features += [ "enable_cast_renderer" ]
   }
 
+  if (is_cobalt && use_starboard_media) {
+    enabled_features += [ "use_starboard_media" ]
+  }
+
   shared_typemaps = [
     {
       types = [
@@ -726,6 +738,30 @@ mojom("mojom") {
     ]
   }
 
+  if (is_cobalt && use_starboard_media) {
+    cpp_typemaps += [
+      {
+        types = [
+          {
+            mojom = "media.mojom.StarboardRendererConfig"
+            cpp = "::media::StarboardRendererConfig"
+          },
+        ]
+        traits_headers = [ "//media/base/ipc/media_param_traits_macros.h" ]
+      },
+      {
+        types = [
+          {
+            mojom = "media.mojom.StarboardRenderingMode"
+            cpp = "::media::StarboardRenderingMode"
+          },
+        ]
+        traits_headers = [ "starboard/starboard_rendering_mode_mojom_traits.h" ]
+        traits_public_deps = [ "//media" ]
+      },
+    ]
+  }
+
   cpp_typemaps += shared_typemaps
   blink_cpp_typemaps = shared_typemaps
 
