diff --git a/media/mojo/services/gpu_mojo_media_client.cc b/media/mojo/services/gpu_mojo_media_client.cc
index 35ed9c8db6825..3f79a71171546 100644
--- a/media/mojo/services/gpu_mojo_media_client.cc
+++ b/media/mojo/services/gpu_mojo_media_client.cc
@@ -106,6 +106,29 @@ VideoDecoderTraits::VideoDecoderTraits(
       android_overlay_factory_cb(std::move(android_overlay_factory_cb)),
       oop_video_decoder(std::move(oop_video_decoder)) {}
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+StarboardRendererTraits::~StarboardRendererTraits() = default;
+StarboardRendererTraits::StarboardRendererTraits(
+    scoped_refptr<base::SequencedTaskRunner> task_runner,
+    mojo::PendingRemote<mojom::MediaLog> media_log_remote,
+    const base::UnguessableToken& overlay_plane_id,
+    base::TimeDelta audio_write_duration_local,
+    base::TimeDelta audio_write_duration_remote,
+    const std::string& max_video_capabilities,
+    mojo::PendingReceiver<mojom::StarboardRendererExtension>
+        renderer_extension_receiver,
+    mojo::PendingRemote<mojom::StarboardRendererClientExtension>
+        client_extension_remote)
+    : task_runner(std::move(task_runner)),
+      media_log_remote(std::move(media_log_remote)),
+      overlay_plane_id(overlay_plane_id),
+      audio_write_duration_local(audio_write_duration_local),
+      audio_write_duration_remote(audio_write_duration_remote),
+      max_video_capabilities(max_video_capabilities),
+      renderer_extension_receiver(std::move(renderer_extension_receiver)),
+      client_extension_remote(std::move(client_extension_remote)) {}
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 GpuMojoMediaClient::GpuMojoMediaClient(
     const gpu::GpuPreferences& gpu_preferences,
     const gpu::GpuDriverBugWorkarounds& gpu_workarounds,
@@ -252,6 +275,26 @@ std::unique_ptr<VideoDecoder> GpuMojoMediaClient::CreateVideoDecoder(
   return CreatePlatformVideoDecoder(traits);
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+std::unique_ptr<Renderer> GpuMojoMediaClient::CreateStarboardRenderer(
+    mojom::FrameInterfaceFactory* /* frame_interfaces */,
+    scoped_refptr<base::SingleThreadTaskRunner> task_runner,
+    mojo::PendingRemote<mojom::MediaLog> media_log_remote,
+    const StarboardRendererConfig& config,
+    mojo::PendingReceiver<mojom::StarboardRendererExtension>
+        renderer_extension_receiver,
+    mojo::PendingRemote<mojom::StarboardRendererClientExtension>
+        client_extension_remote) {
+  StarboardRendererTraits traits(std::move(task_runner),
+      std::move(media_log_remote), config.overlay_plane_id,
+      config.audio_write_duration_local, config.audio_write_duration_remote,
+      config.max_video_capabilities,
+      std::move(renderer_extension_receiver),
+      std::move(client_extension_remote));
+  return CreatePlatformStarboardRenderer(traits);
+}
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 std::unique_ptr<CdmFactory> GpuMojoMediaClient::CreateCdmFactory(
     mojom::FrameInterfaceFactory* frame_interfaces) {
   return CreatePlatformCdmFactory(frame_interfaces);
