diff --git a/media/base/demuxer_stream.h b/media/base/demuxer_stream.h
index d01e266582050..8fd55c248f79b 100644
--- a/media/base/demuxer_stream.h
+++ b/media/base/demuxer_stream.h
@@ -70,6 +70,9 @@ class MEDIA_EXPORT DemuxerStream {
 
   using DecoderBufferVector = std::vector<scoped_refptr<DecoderBuffer>>;
   using ReadCB = base::OnceCallback<void(Status, DecoderBufferVector)>;
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  virtual std::string mime_type() const;
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
 
   // Request buffers to be returned via the provided callback.
   // The first parameter indicates the status of the read request.
