diff --git a/media/renderers/BUILD.gn b/media/renderers/BUILD.gn
index 92fd150eb44a8..19afc5a2b12e6 100644
--- a/media/renderers/BUILD.gn
+++ b/media/renderers/BUILD.gn
@@ -2,11 +2,18 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 source_set("renderers") {
   # Do not expand the visibility here without double-checking with OWNERS, this
   # is a roll-up target which is part of the //media component. Most other DEPs
   # should be using //media and not directly DEP this roll-up target.
   visibility = [ "//media" ]
+  if (is_cobalt && use_starboard_media) {
+    visibility += [ "//media/starboard" ]
+  }
 
   sources = [
     "audio_renderer_impl.cc",
diff --git a/media/test/data/starboard/allocations_1La4QzGeaaQ.txt b/media/test/data/starboard/allocations_1La4QzGeaaQ.txt
new file mode 100644
index 0000000000000..7db6a76f49ccf
