diff --git a/media/mojo/mojom/renderer_extensions.mojom b/media/mojo/mojom/renderer_extensions.mojom
index c326d52bd3090..446c9f44a4096 100644
--- a/media/mojo/mojom/renderer_extensions.mojom
+++ b/media/mojo/mojom/renderer_extensions.mojom
@@ -5,10 +5,14 @@
 module media.mojom;
 
 import "media/mojo/mojom/media_types.mojom";
+import "media/mojo/mojom/video_decoder.mojom";
 import "mojo/public/mojom/base/time.mojom";
 import "mojo/public/mojom/base/unguessable_token.mojom";
 import "ui/gfx/geometry/mojom/geometry.mojom";
 
+[EnableIf=use_starboard_media]
+import "media/mojo/mojom/starboard/starboard_media_types.mojom";
+
 [EnableIf=is_win]
 import "ui/gfx/mojom/buffer_types.mojom";
 // Extension of the mojo::RendererClient communication layer for HLS and Android
@@ -86,6 +90,19 @@ interface MediaFoundationRendererClientExtension {
     gfx.mojom.Size size, mojo_base.mojom.TimeDelta timestamp);
 };
 
+[EnableIf=use_starboard_media]
+// This interface is for StarboardRendererClient to extend
+// mojo:RendererClient as StarboardRendererClient to allow
+// StarboardRenderer (on gpu thread) to post tasks
+// on StarboardRendererClient on renderer thread.
+interface StarboardRendererClientExtension {
+  // Paint a video hole on VideoRendererSink.
+  PaintVideoHoleFrame(gfx.mojom.Size size);
+
+  // Notify the rendering mode from SbPlayer.
+  UpdateStarboardRenderingMode(StarboardRenderingMode mode);
+};
+
 // The Media Foundation Media Engine has two modes for playback.
 // DirectComposition is when the media engine will render directly to a
 // direct composition visual, skipping the Chromium compositor.
@@ -134,6 +151,19 @@ interface MediaFoundationRendererExtension {
   SetMediaFoundationRenderingMode(MediaFoundationRenderingMode mode);
 };
 
+[EnableIf=use_starboard_media]
+// This interface is for StarboardRenderer to extend mojo::Renderer
+// as StarboardRendererWrapper to allow StarboardRendererClient to
+// post tasks on StarboardRendererWrapper on gpu thread.
+interface StarboardRendererExtension {
+  // Subscribe to video geometry changes from
+  // cobalt::media::mojom::VideoGeometryChangeSubscriber
+  OnVideoGeometryChange(gfx.mojom.Rect rect);
+
+  // Pass CommandBufferId to StarboardRendererWrapper.
+  OnGpuChannelTokenReady(CommandBufferId? command_buffer_id);
+};
+
 // This interface is used by the browser to determine if there are any renderers
 // actively using the Media Foundation Renderer. The number of Media Foundation
 // Renderers in use is determined by the number of active connections. The
diff --git a/media/mojo/mojom/starboard/starboard_media_types.mojom b/media/mojo/mojom/starboard/starboard_media_types.mojom
new file mode 100644
index 0000000000000..8c4ea5018e0d6
