diff --git a/media/base/cdm_context.h b/media/base/cdm_context.h
index 260251b42fc69..2818779889a0c 100644
--- a/media/base/cdm_context.h
+++ b/media/base/cdm_context.h
@@ -23,6 +23,9 @@ class ChromeOsCdmContext;
 }
 #endif
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+#include "starboard/drm.h"  // nogncheck
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
 namespace media {
 
 class CallbackRegistration;
@@ -112,6 +115,10 @@ class MEDIA_EXPORT CdmContext {
   virtual MediaCryptoContext* GetMediaCryptoContext();
 #endif
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  virtual SbDrmSystem GetSbDrmSystem();
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 #if BUILDFLAG(IS_FUCHSIA)
   // Returns FuchsiaCdmContext interface when the context is backed by Fuchsia
   // CDM. Otherwise returns nullptr.
