diff --git a/media/filters/source_buffer_stream.h b/media/filters/source_buffer_stream.h
index 8a89cc494867c..19849e85b683c 100644
--- a/media/filters/source_buffer_stream.h
+++ b/media/filters/source_buffer_stream.h
@@ -147,8 +147,8 @@ class MEDIA_EXPORT SourceBufferStream {
   // then base::TimeDelta() is returned.
   base::TimeDelta GetBufferedDuration() const;
 
-  // Returns the size of the buffered data in bytes.
-  size_t GetBufferedSize() const;
+  // Returns the memory usage of the buffered data in bytes.
+  size_t GetMemoryUsage() const;
 
   // Notifies this object that end of stream has been signalled.
   void MarkEndOfStream();
