diff --git a/media/mojo/common/BUILD.gn b/media/mojo/common/BUILD.gn
index ae5303dc1d322..bb9c0fdd2613e 100644
--- a/media/mojo/common/BUILD.gn
+++ b/media/mojo/common/BUILD.gn
@@ -2,6 +2,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 source_set("common") {
   sources = [
     "audio_data_s16_converter.cc",
