diff --git a/media/formats/BUILD.gn b/media/formats/BUILD.gn
index c42bd14703cd5..0dac1c24b37ab 100644
--- a/media/formats/BUILD.gn
+++ b/media/formats/BUILD.gn
@@ -318,6 +318,13 @@ source_set("unit_tests") {
       ]
     }
 
+    if (enable_platform_ac3_eac3_audio) {
+      sources += [
+        "mp4/ac3_unittest.cc",
+        "mp4/eac3_unittest.cc",
+      ]
+    }
+
     if (enable_mse_mpeg2ts_stream_parser) {
       sources += [
         "mp2t/es_adapter_video_unittest.cc",
