diff --git a/media/mojo/clients/mojo_renderer_factory.h b/media/mojo/clients/mojo_renderer_factory.h
index 9b8a5e3686a5e..576868add004f 100644
--- a/media/mojo/clients/mojo_renderer_factory.h
+++ b/media/mojo/clients/mojo_renderer_factory.h
@@ -83,6 +83,18 @@ class MojoRendererFactory final : public RendererFactory {
       VideoRendererSink* video_renderer_sink);
 #endif  // defined (OS_ANDROID)
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  std::unique_ptr<MojoRenderer> CreateStarboardRenderer(
+      mojo::PendingRemote<mojom::MediaLog> media_log_remote,
+      const StarboardRendererConfig& config,
+      mojo::PendingReceiver<mojom::StarboardRendererExtension>
+          renderer_extension_receiver,
+      mojo::PendingRemote<mojom::StarboardRendererClientExtension>
+          client_extension_remote,
+      const scoped_refptr<base::SequencedTaskRunner>& media_task_runner,
+      VideoRendererSink* video_renderer_sink);
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
  private:
   // InterfaceFactory or InterfaceProvider used to create or connect to remote
   // renderer.
