diff --git a/media/capture/BUILD.gn b/media/capture/BUILD.gn
index b56936b1e4296..0a5fd4b304009 100644
--- a/media/capture/BUILD.gn
+++ b/media/capture/BUILD.gn
@@ -254,7 +254,7 @@ component("capture_lib") {
     ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux && !is_cobalt_hermetic_build || is_chromeos) {
     sources += [
       "video/linux/scoped_v4l2_device_fd.cc",
       "video/linux/scoped_v4l2_device_fd.h",
@@ -272,6 +272,11 @@ component("capture_lib") {
     ]
   }
 
+  if (is_cobalt_hermetic_build) {
+    sources +=
+        [ "video/linux/video_capture_device_factory_linux_starboard_stub.cc" ]
+  }
+
   if (is_chromeos_ash) {
     sources += [
       "video/chromeos/camera_3a_controller.cc",
diff --git a/media/capture/video/linux/video_capture_device_factory_linux_starboard_stub.cc b/media/capture/video/linux/video_capture_device_factory_linux_starboard_stub.cc
new file mode 100644
index 0000000000000..6672c9c29248a
