diff --git a/media/mojo/clients/BUILD.gn b/media/mojo/clients/BUILD.gn
index cbd8e90d32a88..02635eb43faaf 100644
--- a/media/mojo/clients/BUILD.gn
+++ b/media/mojo/clients/BUILD.gn
@@ -2,6 +2,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 # Implementations of media C++ interfaces using corresponding mojo services.
 source_set("clients") {
   visibility = [
@@ -102,6 +106,24 @@ source_set("clients") {
 
     deps += [ "//third_party/blink/public:blink_headers" ]
   }
+
+  if (is_cobalt && use_starboard_media) {
+    sources += [
+      "starboard/starboard_renderer_client.cc",
+      "starboard/starboard_renderer_client.h",
+      "starboard/starboard_renderer_client_factory.cc",
+      "starboard/starboard_renderer_client_factory.h",
+    ]
+
+    deps += [
+      "//cobalt/media/service",
+      "//cobalt/media/service/mojom",
+    ]
+
+    visibility += [
+      "//cobalt/renderer",
+    ]
+  }
 }
 
 source_set("unit_tests") {
