diff --git a/media/base/renderer.cc b/media/base/renderer.cc
index 3a2d6a3096c73..ac6185fb5c374 100644
--- a/media/base/renderer.cc
+++ b/media/base/renderer.cc
@@ -33,6 +33,10 @@ std::string GetRendererName(RendererType renderer_type) {
       return "EmbedderDefined";
     case RendererType::kTest:
       return "Media Renderer Implementation For Testing";
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+    case RendererType::kStarboard:
+      return "StarboardRenderer";
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
   }
 }
 
