diff --git a/media/base/BUILD.gn b/media/base/BUILD.gn
index a180a6444c967..68ffb0dd10270 100644
--- a/media/base/BUILD.gn
+++ b/media/base/BUILD.gn
@@ -14,6 +14,9 @@ import("//testing/libfuzzer/fuzzer_test.gni")
 if (is_android) {
   import("//build/config/android/rules.gni")
 }
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
 
 source_set("base") {
   # Do not expand the visibility here without double-checking with OWNERS, this
@@ -459,7 +462,15 @@ source_set("base") {
     ]
   }
 
-  if (is_android) {
+  if (is_cobalt && use_starboard_media) {
+    sources += [
+        "starboard/demuxer_memory_limit_starboard.cc",
+        "starboard/renderer_factory_traits.h",
+        "starboard/starboard_renderer_config.cc",
+        "starboard/starboard_renderer_config.h",
+        "starboard/starboard_rendering_mode.h"
+    ]
+  } else if (is_android) {
     sources += [ "demuxer_memory_limit_android.cc" ]
   } else if (is_castos) {
     sources += [ "demuxer_memory_limit_cast.cc" ]
@@ -478,6 +489,12 @@ source_set("base") {
       "media_drm_storage.h",
     ]
   }
+
+  if (is_cobalt && use_starboard_media) {
+    deps += [
+      "//starboard:starboard_group",
+    ]
+  }
 }
 
 source_set("video_facing") {
@@ -678,6 +695,12 @@ source_set("unit_tests") {
   if (is_castos) {
     sources += [ "demuxer_memory_limit_cast_unittest.cc" ]
   }
+
+  if (is_cobalt && use_starboard_media) {
+    sources += [
+      "starboard/starboard_renderer_config_unittest.cc"
+    ]
+  }
 }
 
 source_set("perftests") {
