diff --git a/media/formats/mp4/box_definitions.cc b/media/formats/mp4/box_definitions.cc
index 2cded6670f5e4..01db74ed694d7 100644
--- a/media/formats/mp4/box_definitions.cc
+++ b/media/formats/mp4/box_definitions.cc
@@ -1624,11 +1624,13 @@ bool AudioSampleEntry::Parse(BoxReader* reader) {
 #endif  // BUILDFLAG(ENABLE_PLATFORM_DTS_AUDIO)
 
 #if BUILDFLAG(ENABLE_PLATFORM_AC3_EAC3_AUDIO)
-  if (format == FOURCC_AC3) {
+  if (format == FOURCC_AC3 ||
+      (format == FOURCC_ENCA && sinf.format.format == FOURCC_AC3)) {
     RCHECK_MEDIA_LOGGED(reader->ReadChild(&ac3), reader->media_log(),
                         "Failure parsing AC3SpecificBox (dac3)");
   }
-  if (format == FOURCC_EAC3) {
+  if (format == FOURCC_EAC3 ||
+      (format == FOURCC_ENCA && sinf.format.format == FOURCC_EAC3)) {
     RCHECK_MEDIA_LOGGED(reader->ReadChild(&eac3), reader->media_log(),
                         "Failure parsing EC3SpecificBox (dec3)");
   }
