diff --git a/media/media_options.gni b/media/media_options.gni
index cdc3e93753397..fcc6ecca4752c 100644
--- a/media/media_options.gni
+++ b/media/media_options.gni
@@ -13,6 +13,10 @@ import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/libaom/options.gni")
 import("//third_party/libgav1/options.gni")
 
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
+
 # This flag sets defaults for the current generation of cast devices.
 is_cast_media_device = is_castos || is_cast_android
 
@@ -345,6 +349,10 @@ media_subcomponent_deps = [
   "//media/video",
 ]
 
+if (is_cobalt && use_starboard_media) {
+  media_subcomponent_deps += [ "//media/starboard" ]
+}
+
 if (is_fuchsia) {
   media_subcomponent_deps += [ "//media/fuchsia/common" ]
 }
