diff --git a/net/disk_cache/simple/simple_version_upgrade.cc b/net/disk_cache/simple/simple_version_upgrade.cc
index 09d80c5af8722..9c898e4079c3b 100644
--- a/net/disk_cache/simple/simple_version_upgrade.cc
+++ b/net/disk_cache/simple/simple_version_upgrade.cc
@@ -34,8 +34,15 @@ void LogMessageFailedUpgradeFromVersion(int version) {
 
 bool WriteFakeIndexFile(disk_cache::BackendFileOperations* file_operations,
                         const base::FilePath& file_name) {
+#if BUILDFLAG(IS_STARBOARD)
+  // Overwrites existing fake index files in case they were not deleted in
+  // previous cache startup. We can consider upstream this change.
+  base::File file = file_operations->OpenFile(
+      file_name, base::File::FLAG_CREATE_ALWAYS | base::File::FLAG_WRITE);
+#else
   base::File file = file_operations->OpenFile(
       file_name, base::File::FLAG_CREATE | base::File::FLAG_WRITE);
+#endif
   if (!file.IsValid())
     return false;
 
