diff --git a/net/base/network_change_notifier.cc b/net/base/network_change_notifier.cc
index 953c4ff32adb9..0847d5cc0da5f 100644
--- a/net/base/network_change_notifier.cc
+++ b/net/base/network_change_notifier.cc
@@ -31,7 +31,9 @@
 #include "third_party/abseil-cpp/absl/types/optional.h"
 #include "url/gurl.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_STARBOARD)
+#include "net/base/network_change_notifier_starboard.h"
+#elif BUILDFLAG(IS_WIN)
 #include "net/base/network_change_notifier_win.h"
 #elif BUILDFLAG(IS_LINUX)
 #include "net/base/network_change_notifier_linux.h"
@@ -307,7 +309,9 @@ std::unique_ptr<NetworkChangeNotifier> NetworkChangeNotifier::CreateIfNeeded(
         initial_type, initial_subtype);
   }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_STARBOARD)
+return std::make_unique<NetworkChangeNotifierStarboard>();
+#elif BUILDFLAG(IS_WIN)
   std::unique_ptr<NetworkChangeNotifierWin> network_change_notifier =
       std::make_unique<NetworkChangeNotifierWin>();
   network_change_notifier->WatchForAddressChange();
