diff --git a/services/network/network_change_manager.h b/services/network/network_change_manager.h
index 4d460023efa38..00de773079d7a 100644
--- a/services/network/network_change_manager.h
+++ b/services/network/network_change_manager.h
@@ -18,7 +18,7 @@
 #include "net/base/network_change_notifier.h"
 #include "services/network/public/mojom/network_change_manager.mojom.h"
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 #include "services/network/public/mojom/network_interface_change_listener.mojom.h"
 #endif
 
@@ -31,7 +31,7 @@ namespace network {
 class COMPONENT_EXPORT(NETWORK_SERVICE) NetworkChangeManager
     : public mojom::NetworkChangeManager,
       public net::NetworkChangeNotifier::NetworkChangeObserver
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
     ,
       public mojom::NetworkInterfaceChangeListener
 #endif
@@ -66,7 +66,7 @@ class COMPONENT_EXPORT(NETWORK_SERVICE) NetworkChangeManager
       mojom::ConnectionSubtype new_connection_subtype) override;
 #endif
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
   void BindNetworkInterfaceChangeListener(
       mojo::PendingAssociatedReceiver<mojom::NetworkInterfaceChangeListener>)
       override;
@@ -88,7 +88,7 @@ class COMPONENT_EXPORT(NETWORK_SERVICE) NetworkChangeManager
 
   std::unique_ptr<net::NetworkChangeNotifier> network_change_notifier_;
   mojo::ReceiverSet<mojom::NetworkChangeManager> receivers_;
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
   mojo::AssociatedReceiver<mojom::NetworkInterfaceChangeListener>
       interface_change_listener_receiver_{this};
 #endif
