diff --git a/services/network/network_service.cc b/services/network/network_service.cc
index 54d2854338f37..7f2606c0f0070 100644
--- a/services/network/network_service.cc
+++ b/services/network/network_service.cc
@@ -390,7 +390,7 @@ void NetworkService::Initialize(mojom::NetworkServiceParamsPtr params,
               params->initial_connection_subtype),
           mock_network_change_notifier);
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
   if (params->initial_address_map) {
     // The NetworkChangeNotifierPassive should only be included if it's
     // necessary to instantiate an AddressMapCacheLinux rather than an
