diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index ea7cfd9e31791..5f3d35eeeec16 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -436,7 +436,7 @@ component("skia") {
     ]
   }
 
-  if (is_linux || is_chromeos || is_android) {
+  if (is_linux && !is_starboard || is_chromeos || is_android) {
     sources += [
       # Retain the files for the SkFontMgr_Android on linux to emulate android
       # fonts. See content/zygote/zygote_main_linux.cc
