diff --git a/testing/test.gni b/testing/test.gni
index ea121c365e474..cb8164dad8aad 100644
--- a/testing/test.gni
+++ b/testing/test.gni
@@ -13,6 +13,9 @@ import("//build/config/gclient_args.gni")
 import("//build/config/rts.gni")
 import("//build/rust/rust_static_library.gni")
 import("//build_overrides/build.gni")
+if (is_starboard) {
+  import("//cobalt/build/modular_executable.gni")
+}
 
 declare_args() {
   # Some component repos (e.g. ANGLE) import //testing but do not have
@@ -957,6 +960,15 @@ template("test") {
         data_deps += [ ":${invoker.target_name}__rts_filters" ]
       }
     }
+    # TODO: b/384652502 - Cobalt: Remove use_custom_libc after setting
+    # is_starboard as a toolchain arg.
+  } else if (is_starboard && use_custom_libc) {
+    modular_executable(target_name) {
+      forward_variables_from(invoker,
+                             "*",
+                             TESTONLY_AND_VISIBILITY + [ "use_xvfb" ])
+      testonly = true
+    }
   } else if (!is_nacl) {
     if (is_mac || is_win) {
       assert(!defined(invoker.use_xvfb) || !invoker.use_xvfb)
@@ -1168,7 +1180,10 @@ template("isolated_script_test") {
 
 # Test defaults.
 set_defaults("test") {
-  if (is_android) {
+  if (is_cobalt_hermetic_build) {
+    configs = default_shared_library_configs
+    configs -= [ "//build/config/compiler:no_unresolved_symbols" ]
+  } else if (is_android) {
     configs = default_shared_library_configs
     configs -= [ "//build/config/android:hide_all_but_jni_onload" ]
     configs += [ "//build/config/android:hide_all_but_jni" ]
