diff --git a/third_party/blink/renderer/platform/network/mime/mime_type_registry.cc b/third_party/blink/renderer/platform/network/mime/mime_type_registry.cc
index ac6faa8fdcdc1..8a05039b8fabb 100644
--- a/third_party/blink/renderer/platform/network/mime/mime_type_registry.cc
+++ b/third_party/blink/renderer/platform/network/mime/mime_type_registry.cc
@@ -18,6 +18,9 @@
 #include "third_party/blink/renderer/platform/wtf/assertions.h"
 #include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
 
+// For BUILDFLAG(USE_STARBOARD_MEDIA)
+#include "build/build_config.h"
+
 namespace blink {
 
 namespace {
@@ -133,7 +136,11 @@ bool MIMETypeRegistry::IsSupportedMediaMIMEType(const String& mime_type,
 MIMETypeRegistry::SupportsType MIMETypeRegistry::SupportsMediaMIMEType(
     const String& mime_type,
     const String& codecs) {
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  const std::string ascii_mime_type = mime_type.Ascii();
+#else   // BUILDFLAG(USE_STARBOARD_MEDIA)
   const std::string ascii_mime_type = ToLowerASCIIOrEmpty(mime_type);
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
   std::vector<std::string> codec_vector;
   media::SplitCodecs(ToASCIIOrEmpty(codecs), &codec_vector);
   return static_cast<SupportsType>(
