diff --git a/third_party/blink/public/platform/web_content_decryption_module.h b/third_party/blink/public/platform/web_content_decryption_module.h
index 3e42dcb34e7a0..d5e8977132e1e 100644
--- a/third_party/blink/public/platform/web_content_decryption_module.h
+++ b/third_party/blink/public/platform/web_content_decryption_module.h
@@ -53,6 +53,10 @@ class BLINK_PLATFORM_EXPORT WebContentDecryptionModule {
 
   virtual void GetStatusForPolicy(const WebString& min_hdcp_version,
                                   WebContentDecryptionModuleResult) = 0;
+
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  virtual bool GetMetrics(std::string& metrics_results) = 0;
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
 };
 
 }  // namespace blink
