diff --git a/third_party/crashpad/crashpad/client/crashpad_info.cc b/third_party/crashpad/crashpad/client/crashpad_info.cc
index cd6f234818bae..7f311d078f85d 100644
--- a/third_party/crashpad/crashpad/client/crashpad_info.cc
+++ b/third_party/crashpad/crashpad/client/crashpad_info.cc
@@ -97,6 +97,9 @@ extern "C" int* CRASHPAD_NOTE_REFERENCE;
 CrashpadInfo* CrashpadInfo::GetCrashpadInfo() {
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID) || \
     BUILDFLAG(IS_FUCHSIA)
+// TODO: b/406511608 - Cobalt: try removing this guard once the hermetic build
+// of crashpad includes the crashpad/client/crashpad_info_note.S souce file.
+#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
   // This otherwise-unused reference is used so that any module that
   // references GetCrashpadInfo() will also include the note in the
   // .note.crashpad.info section. That note in turn contains the address of
@@ -104,6 +107,7 @@ CrashpadInfo* CrashpadInfo::GetCrashpadInfo() {
   // structure without requiring the use of the dynamic symbol table.
   static volatile int* pointer_to_note_section = CRASHPAD_NOTE_REFERENCE;
   (void)pointer_to_note_section;
+#endif  // !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 #endif
   return &g_crashpad_info;
 }
