diff --git a/third_party/blink/renderer/platform/media/web_source_buffer_impl.h b/third_party/blink/renderer/platform/media/web_source_buffer_impl.h
index 5ecfab637271d..25c272f8d1f57 100644
--- a/third_party/blink/renderer/platform/media/web_source_buffer_impl.h
+++ b/third_party/blink/renderer/platform/media/web_source_buffer_impl.h
@@ -24,6 +24,10 @@ enum class SourceBufferParseWarning;
 
 namespace blink {
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+class ExceptionState;
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 class PLATFORM_EXPORT WebSourceBufferImpl : public WebSourceBuffer {
  public:
   WebSourceBufferImpl(const std::string& id, media::ChunkDemuxer* demuxer);
@@ -57,6 +61,11 @@ class PLATFORM_EXPORT WebSourceBufferImpl : public WebSourceBuffer {
   void SetAppendWindowEnd(double end) override;
   void RemovedFromMediaSource() override;
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  // Return the highest presentation timestamp written to the Renderer.
+  double GetWriteHead(ExceptionState& exception_state) const override;
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
  private:
   // Demuxer callback handler to process an initialization segment received
   // during an append() call.
