diff --git a/third_party/blink/renderer/platform/media/web_content_decryption_module_impl.cc b/third_party/blink/renderer/platform/media/web_content_decryption_module_impl.cc
index 3af2b95bad431..c6a533b488077 100644
--- a/third_party/blink/renderer/platform/media/web_content_decryption_module_impl.cc
+++ b/third_party/blink/renderer/platform/media/web_content_decryption_module_impl.cc
@@ -154,6 +154,29 @@ void WebContentDecryptionModuleImpl::GetStatusForPolicy(
           kGetStatusForPolicyUMAName));
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+bool WebContentDecryptionModuleImpl::GetMetrics(std::string& metrics_results) {
+  auto cdm_context_ref = adapter_->GetCdmContextRef();
+  DCHECK(cdm_context_ref);
+
+  auto* cdm_context = cdm_context_ref->GetCdmContext();
+  DCHECK(cdm_context);
+
+  auto sb_drm = cdm_context->GetSbDrmSystem();
+  DCHECK(SbDrmSystemIsValid(sb_drm));
+
+  int size = 0;
+  const uint8_t* raw_metrics =
+      static_cast<const uint8_t*>(SbDrmGetMetrics(sb_drm, &size));
+  if (!raw_metrics || size < 0) {
+    return false;
+  }
+
+  metrics_results.assign(raw_metrics, raw_metrics + size);
+  return true;
+}
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 std::unique_ptr<media::CdmContextRef>
 WebContentDecryptionModuleImpl::GetCdmContextRef() {
   return adapter_->GetCdmContextRef();
