diff --git a/third_party/blink/renderer/modules/mediasource/source_buffer.h b/third_party/blink/renderer/modules/mediasource/source_buffer.h
index 0178bf1a8e66c..0ac28dca70f0b 100644
--- a/third_party/blink/renderer/modules/mediasource/source_buffer.h
+++ b/third_party/blink/renderer/modules/mediasource/source_buffer.h
@@ -113,6 +113,12 @@ class SourceBuffer final : public EventTargetWithInlineData,
   AudioTrackList& audioTracks();
   VideoTrackList& videoTracks();
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  // Cobalt-specific method that returns the highest presentation
+  // timestamp written to the Renderer.
+  double GetWriteHead(ExceptionState& exception_state) const;
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
   // "_Locked" requires these be called while in the scope of callback of
   // |source_|'s attachment's RunExclusively(). Other methods without "_Locked"
   // may also require the same, since they can be called from within these
