diff --git a/third_party/expat/BUILD.gn b/third_party/expat/BUILD.gn
index 7699d1c35006d..193c5968870ef 100644
--- a/third_party/expat/BUILD.gn
+++ b/third_party/expat/BUILD.gn
@@ -9,7 +9,7 @@ import("//testing/libfuzzer/fuzzer_test.gni")
 # let's not pull it in twice.
 # Chromecast doesn't ship expat as a system library.
 # Libfuzzer and AFL need to build library from sources.
-if (((is_linux && !is_castos) || is_chromeos) && !use_fuzzing_engine) {
+if (((is_linux && !is_castos) || is_chromeos) && !use_fuzzing_engine && !is_starboard) {
   config("expat_config") {
     libs = [ "expat" ]
   }
