diff --git a/third_party/fontconfig/include/config.h b/third_party/fontconfig/include/config.h
index 61f8a0c6fe090..d69859a037d02 100644
--- a/third_party/fontconfig/include/config.h
+++ b/third_party/fontconfig/include/config.h
@@ -4,6 +4,8 @@
 /* Define if building universal (internal helper macro) */
 /* #undef AC_APPLE_UNIVERSAL_BUILD */
 
+#include "build/build_config.h"
+
 /* The normal alignment of `double', in bytes. */
 #define ALIGNOF_DOUBLE 8
 
@@ -155,8 +157,13 @@
 /* Define to 1 if you have the `random' function. */
 #define HAVE_RANDOM 1
 
+#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
+/* TODO: (cobalt b/398295440) Add `random_r' support to Evergreen. */
+#define HAVE_RANDOM_R 0
+#else
 /* Define to 1 if you have the `random_r' function. */
 #define HAVE_RANDOM_R 1
+#endif
 
 /* Define to 1 if you have the `rand_r' function. */
 #define HAVE_RAND_R 1
