diff --git a/third_party/blink/renderer/platform/media/web_source_buffer_impl.cc b/third_party/blink/renderer/platform/media/web_source_buffer_impl.cc
index a040298abd08f..a3eb88f63b349 100644
--- a/third_party/blink/renderer/platform/media/web_source_buffer_impl.cc
+++ b/third_party/blink/renderer/platform/media/web_source_buffer_impl.cc
@@ -245,6 +245,14 @@ void WebSourceBufferImpl::RemovedFromMediaSource() {
   client_ = nullptr;
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+double WebSourceBufferImpl::GetWriteHead(
+    ExceptionState& exception_state) const {
+  DCHECK(demuxer_);
+  return demuxer_->GetWriteHead(id_).InSecondsF();
+}
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 WebMediaPlayer::TrackType mediaTrackTypeToBlink(media::MediaTrack::Type type) {
   switch (type) {
     case media::MediaTrack::Audio:
