diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 8e8bb74489bd6..2a8e6d11e19a5 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -98,7 +98,9 @@ dependencies {
     compile "com.android.support:viewpager:${supportLibVersion}"
 
     compile 'com.android.support:multidex:1.0.0'
-
+    
+    compile 'com.google.android.gms:play-services-ads-identifier:18.2.0'
+    
     compile 'com.google.code.findbugs:jsr305:3.0.2'
     compile 'com.google.firebase:firebase-iid:21.0.1'
     compile 'com.google.firebase:firebase-messaging:21.0.1'
