diff --git a/third_party/blink/public/platform/web_media_player_client.h b/third_party/blink/public/platform/web_media_player_client.h
index 71442dd9fde90..96dcf265739e1 100644
--- a/third_party/blink/public/platform/web_media_player_client.h
+++ b/third_party/blink/public/platform/web_media_player_client.h
@@ -232,6 +232,11 @@ class BLINK_PLATFORM_EXPORT WebMediaPlayerClient {
   // Notify the client that the RemotePlayback has been disabled/enabled.
   virtual void OnRemotePlaybackDisabled(bool disabled) = 0;
 
+  // Returns the DOMNodeId of the DOM element hosting this media player.
+  virtual int GetElementId() = 0;
+
+  virtual std::string getMaxVideoCapabilities() const {return "";}
+
  protected:
   ~WebMediaPlayerClient() = default;
 };
