diff --git a/third_party/rust/Cargo.toml b/third_party/rust/Cargo.toml
index e7e61ce21a044..8652d39796afa 100644
--- a/third_party/rust/Cargo.toml
+++ b/third_party/rust/Cargo.toml
@@ -10,22 +10,16 @@ members = ["cxxbridge_cmd/v1/crate"]
 
 [dependencies]
 bitflags = "1"
-cargo-platform = "0.1"
-cargo_metadata = "0.14"
-clap = "4"
 cxx = "1"
 cxxbridge-cmd = "1"
 lazy_static = "1"
-once_cell = "1"
 proc-macro2 = "1"
 quote = "1"
 rstest = "0.12"
-semver = "1"
 serde = "1"
 serde_json = "1"
 small_ctor = "0.1"
 static_assertions = "1"
-tempfile = "3"
 
 [dependencies.memoffset]
 version = "0.6"
@@ -43,10 +37,6 @@ features = ["unbounded_depth", "float_roundtrip"]
 version = "1"
 features = ["full"]
 
-[dependencies.toml]
-version = "0.5"
-features = ["preserve_order"]
-
 [dependencies.unicode-linebreak]
 version = "0.1"
 build-script-outputs = ["tables.rs"]
@@ -54,10 +44,6 @@ build-script-outputs = ["tables.rs"]
 path = "aho_corasick/v0_7/crate"
 package = "aho-corasick"
 
-[patch.crates-io.atty_v0_2]
-path = "atty/v0_2/crate"
-package = "atty"
-
 [patch.crates-io.autocfg_v1]
 path = "autocfg/v1/crate"
 package = "autocfg"
@@ -66,18 +52,6 @@ package = "autocfg"
 path = "bitflags/v1/crate"
 package = "bitflags"
 
-[patch.crates-io.camino_v1]
-path = "camino/v1/crate"
-package = "camino"
-
-[patch.crates-io.cargo-platform_v0_1]
-path = "cargo_platform/v0_1/crate"
-package = "cargo-platform"
-
-[patch.crates-io.cargo_metadata_v0_14]
-path = "cargo_metadata/v0_14/crate"
-package = "cargo_metadata"
-
 [patch.crates-io.cfg-if_v1]
 path = "cfg_if/v1/crate"
 package = "cfg-if"
@@ -106,18 +80,6 @@ package = "cxxbridge-cmd"
 path = "cxxbridge_macro/v1/crate"
 package = "cxxbridge-macro"
 
-[patch.crates-io.fastrand_v1]
-path = "fastrand/v1/crate"
-package = "fastrand"
-
-[patch.crates-io.hashbrown_v0_11]
-path = "hashbrown/v0_11/crate"
-package = "hashbrown"
-
-[patch.crates-io.indexmap_v1]
-path = "indexmap/v1/crate"
-package = "indexmap"
-
 [patch.crates-io.itoa_v0_4]
 path = "itoa/v0_4/crate"
 package = "itoa"
@@ -146,10 +108,6 @@ package = "memchr"
 path = "memoffset/v0_6/crate"
 package = "memoffset"
 
-[patch.crates-io.once_cell_v1]
-path = "once_cell/v1/crate"
-package = "once_cell"
-
 [patch.crates-io.os_str_bytes_v6]
 path = "os_str_bytes/v6/crate"
 package = "os_str_bytes"
@@ -170,10 +128,6 @@ package = "regex-syntax"
 path = "regex/v1/crate"
 package = "regex"
 
-[patch.crates-io.remove_dir_all_v0_5]
-path = "remove_dir_all/v0_5/crate"
-package = "remove_dir_all"
-
 [patch.crates-io.rstest_v0_12]
 path = "rstest/v0_12/crate"
 package = "rstest"
@@ -226,18 +180,10 @@ package = "strsim"
 path = "syn/v1/crate"
 package = "syn"
 
-[patch.crates-io.tempfile_v3]
-path = "tempfile/v3/crate"
-package = "tempfile"
-
 [patch.crates-io.termcolor_v1]
 path = "termcolor/v1/crate"
 package = "termcolor"
 
-[patch.crates-io.toml_v0_5]
-path = "toml/v0_5/crate"
-package = "toml"
-
 [patch.crates-io.unicode-ident_v1]
 path = "unicode_ident/v1/crate"
 package = "unicode-ident"
