diff --git a/third_party/liblouis/README.chromium b/third_party/liblouis/README.chromium
index d8f95c61ac518..8811e68a8172f 100644
--- a/third_party/liblouis/README.chromium
+++ b/third_party/liblouis/README.chromium
@@ -2,7 +2,7 @@ Name: Braille Translation Library
 Short Name: liblouis
 URL: https://github.com/liblouis/liblouis
 Version: 3.22.0
-CPEPrefix: cpe:/a:liblouis:liblouis:3.12.0
+CPEPrefix: cpe:/a:liblouis:liblouis:3.22.0
 License: LGPL3 and GPL3
 License Android Compatible: No
 License File: LICENSE
