diff --git a/third_party/blink/renderer/platform/network/http_parsers.cc b/third_party/blink/renderer/platform/network/http_parsers.cc
index 3d329ce327f21..c6af57f2456a0 100644
--- a/third_party/blink/renderer/platform/network/http_parsers.cc
+++ b/third_party/blink/renderer/platform/network/http_parsers.cc
@@ -176,7 +176,11 @@ blink::CSPSourceListPtr ConvertToBlink(const CSPSourceListPtr& source_list) {
       source_list->allow_inline_speculation_rules, source_list->allow_eval,
       source_list->allow_wasm_eval, source_list->allow_wasm_unsafe_eval,
       source_list->allow_dynamic, source_list->allow_unsafe_hashes,
+#if BUILDFLAG(IS_COBALT)
+      source_list->report_sample, source_list->cobalt_insecure_local_network);
+#else
       source_list->report_sample);
+#endif
 }
 
 blink::ContentSecurityPolicyHeaderPtr ConvertToBlink(
