diff --git a/third_party/blink/renderer/modules/mediasource/BUILD.gn b/third_party/blink/renderer/modules/mediasource/BUILD.gn
index 513d0e9d92b73..d4727974c0557 100644
--- a/third_party/blink/renderer/modules/mediasource/BUILD.gn
+++ b/third_party/blink/renderer/modules/mediasource/BUILD.gn
@@ -4,6 +4,9 @@
 
 import("//media/media_options.gni")
 import("//third_party/blink/renderer/modules/modules.gni")
+if (is_cobalt) {
+  import("//starboard/build/buildflags.gni")
+}
 
 blink_modules_sources("mediasource") {
   sources = [
@@ -52,4 +55,7 @@ blink_modules_sources("mediasource") {
     # Ensure the generated webcodecs v8 config and chunk bindings are available.
     "//third_party/blink/renderer/modules/webcodecs:webcodecs",
   ]
+  if (is_cobalt && use_starboard_media) {
+    deps += [ "//starboard:starboard_group" ]
+  }
 }
