diff --git a/third_party/libevent/BUILD.gn b/third_party/libevent/BUILD.gn
index 61e94cc0cd03f..df803a64cb746 100644
--- a/third_party/libevent/BUILD.gn
+++ b/third_party/libevent/BUILD.gn
@@ -36,7 +36,7 @@ static_library("libevent") {
   ]
 
   defines = [ "HAVE_CONFIG_H" ]
-  if (is_cronet_build) {
+  if (is_cronet_build || is_starboard) {
     include_dirs = [ "compat" ]
   } else {
     include_dirs = []
