diff --git a/ui/gl/gl_switches.cc b/ui/gl/gl_switches.cc
index 920de2874970f..8e223203abeee 100644
--- a/ui/gl/gl_switches.cc
+++ b/ui/gl/gl_switches.cc
@@ -12,7 +12,9 @@
 #endif
 
 #if BUILDFLAG(IS_LINUX)
+#if !BUILDFLAG(IS_STARBOARD)
 #include <vulkan/vulkan_core.h>
+#endif  // !BUILDFLAG(IS_STARBOARD)
 #include "third_party/angle/src/gpu_info_util/SystemInfo.h"  // nogncheck
 #endif
 
@@ -271,7 +273,7 @@ bool IsDefaultANGLEVulkan() {
       base::android::SDK_VERSION_Q)
     return false;
 #endif  // BUILDFLAG(IS_ANDROID)
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD)
   angle::SystemInfo system_info;
   if (!angle::GetSystemInfoVulkan(&system_info))
     return false;
