diff --git a/ui/gl/BUILD.gn b/ui/gl/BUILD.gn
index 761888bf4a540..f68907405e914 100644
--- a/ui/gl/BUILD.gn
+++ b/ui/gl/BUILD.gn
@@ -252,7 +252,7 @@ component("gl") {
 
   if (use_ozone) {
     deps += [ "//ui/ozone:buildflags" ]
-    if (use_egl && !is_fuchsia) {
+    if (use_egl && !is_fuchsia && !is_cobalt_hermetic_build) {
       data_deps += [
         "//third_party/angle:libEGL",
         "//third_party/angle:libGLESv2",
