diff --git a/ui/ozone/platform/wayland/test/test_wayland_server_thread.h b/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
index eb7d89e944bf4..1d4ce3bee5c5c 100644
--- a/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
+++ b/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
@@ -12,7 +12,7 @@
 #include <vector>
 
 #include "base/memory/raw_ptr.h"
-#include "base/message_loop/message_pump_libevent.h"
+#include "base/message_loop/message_pump_epoll.h"
 #include "base/threading/thread.h"
 #include "base/threading/thread_checker.h"
 #include "ui/display/types/display_constants.h"
@@ -84,8 +84,9 @@ struct TestServerListener {
 
 class TestSelectionDeviceManager;
 
-class TestWaylandServerThread : public base::Thread,
-                                base::MessagePumpLibevent::FdWatcher {
+class TestWaylandServerThread : public TestOutput::Delegate,
+                                public base::Thread,
+                                base::MessagePumpEpoll::FdWatcher {
  public:
   class OutputDelegate;
 
@@ -204,7 +205,7 @@ class TestWaylandServerThread : public base::Thread,
   // server's thread.
   void DoRun(base::OnceClosure closure);
 
-  // base::MessagePumpLibevent::FdWatcher
+  // base::MessagePumpEpoll::FdWatcher
   void OnFileCanReadWithoutBlocking(int fd) override;
   void OnFileCanWriteWithoutBlocking(int fd) override;
 
@@ -253,7 +254,7 @@ class TestWaylandServerThread : public base::Thread,
 
   std::vector<std::unique_ptr<GlobalObject>> globals_;
 
-  base::MessagePumpLibevent::FdWatchController controller_;
+  base::MessagePumpEpoll::FdWatchController controller_;
 
   raw_ptr<OutputDelegate> output_delegate_ = nullptr;
 
