diff --git a/ui/events/ozone/evdev/touch_event_converter_evdev.h b/ui/events/ozone/evdev/touch_event_converter_evdev.h
index 41d7c3126a15e..fab4948e5160b 100644
--- a/ui/events/ozone/evdev/touch_event_converter_evdev.h
+++ b/ui/events/ozone/evdev/touch_event_converter_evdev.h
@@ -23,7 +23,7 @@
 #include "base/component_export.h"
 #include "base/files/file_path.h"
 #include "base/files/scoped_file.h"
-#include "base/message_loop/message_pump_libevent.h"
+#include "base/message_loop/message_pump_epoll.h"
 #include "base/metrics/field_trial_params.h"
 #include "base/time/time.h"
 #include "ui/events/ozone/evdev/event_converter_evdev.h"
@@ -98,7 +98,7 @@ class COMPONENT_EXPORT(EVDEV) TouchEventConverterEvdev
  private:
   friend class MockTouchEventConverterEvdev;
 
-  // Overidden from base::MessagePumpLibevent::FdWatcher.
+  // Overridden from base::MessagePumpEpoll::FdWatcher.
   void OnFileCanReadWithoutBlocking(int fd) override;
 
   virtual void Reinitialize();
