diff --git a/ui/events/keycodes/keyboard_codes_posix.h b/ui/events/keycodes/keyboard_codes_posix.h
index 22ef5478a902c..309c395cc7fe0 100644
--- a/ui/events/keycodes/keyboard_codes_posix.h
+++ b/ui/events/keycodes/keyboard_codes_posix.h
@@ -31,6 +31,8 @@
 #ifndef UI_EVENTS_KEYCODES_KEYBOARD_CODES_POSIX_H_
 #define UI_EVENTS_KEYCODES_KEYBOARD_CODES_POSIX_H_
 
+#include "build/build_config.h"
+
 namespace ui {
 
 // When adding a new KeyboardCode, be sure to also update the associated mojom
@@ -253,6 +255,10 @@ enum KeyboardCode {
   VKEY_DICTATE = 0xEE,
   // All applications - this also triggers the launcher in Chrome OS.
   VKEY_ALL_APPLICATIONS = 0xEF,
+#if BUILDFLAG(IS_COBALT)
+  // Toggle subtitles
+  KEY_SUBTITLES = 0x1CC,
+#endif  // BUILDFLAG(IS_COBALT)
 };
 
 }  // namespace ui
