diff --git a/ui/events/keycodes/keyboard_code_conversion_android.cc b/ui/events/keycodes/keyboard_code_conversion_android.cc
index 252b7c2aa3164..7f38f4874a682 100644
--- a/ui/events/keycodes/keyboard_code_conversion_android.cc
+++ b/ui/events/keycodes/keyboard_code_conversion_android.cc
@@ -771,6 +771,10 @@ KeyboardCode KeyboardCodeFromAndroidKeyCode(int keycode) {
       return VKEY_PRIOR;
     case AKEYCODE_CHANNEL_DOWN:
       return VKEY_NEXT;
+#if BUILDFLAG(IS_COBALT)
+    case AKEYCODE_CAPTIONS:
+      return KEY_SUBTITLES;
+#endif  // BUILDFLAG(IS_COBALT)
     default:
       return VKEY_UNKNOWN;
   }
