diff --git a/ui/file_manager/image_loader/piex/Makefile b/ui/file_manager/image_loader/piex/Makefile
index 3e0204f93c5ab..c1177cb4aaa0a 100644
--- a/ui/file_manager/image_loader/piex/Makefile
+++ b/ui/file_manager/image_loader/piex/Makefile
@@ -47,5 +47,5 @@ release:
 	$(shell cp piex*wasm extension)
 
 clean:
-	$(shell rm -f tests.{result*,log} package-lock* *.bc piex.out.js)
+	$(shell rm -f tests.{result*,log} *.bc piex.out.js)
 	$(shell rm -f piex*wasm extension/piex*wasm)
